src/Resources/dev/shop/payment/integrated.js

```diff
@@ -0,0 +1,180 @@
+    document.querySelector('.payplugIntegratedPayment').classList.add('payplugIntegratedPayment--loaded');
+    document.querySelector('#next-step').classList.add('disabled');
+    if (null === IntegratedPayment.options.api) {
+      IntegratedPayment.load();
+    }
+  },
+  closeFields() {
+    document.querySelector('.payplugIntegratedPayment').classList.remove('payplugIntegratedPayment--loaded');
+    document.querySelector('#next-step').classList.remove('disabled');
+  },
+  load() {
+    IntegratedPayment.options.api = integratedPaymentApi = new Payplug.IntegratedPayment(true);
+    integratedPaymentApi.setDisplayMode3ds(Payplug.DisplayMode3ds.LIGHTBOX);
+    IntegratedPayment.options.form.cardHolder = integratedPaymentApi.cardHolder(
+      document.querySelector('.cardHolder-input-container'),
+      {
+        default: IntegratedPayment.options.inputStyle.default,
+        placeholder: payplug_integrated_payment_params.cardholder
+      }
+    );
+    IntegratedPayment.options.form.pan = integratedPaymentApi.cardNumber(
+      document.querySelector('.pan-input-container'),
+      {
+        default: IntegratedPayment.options.inputStyle.default,
+        placeholder: payplug_integrated_payment_params.pan
+      }
+    );
+    IntegratedPayment.options.form.cvv = integratedPaymentApi.cvv(
+      document.querySelector('.cvv-input-container'),
+      {
+        default: IntegratedPayment.options.inputStyle.default,
+        placeholder: payplug_integrated_payment_params.cvv
+      }
+    );
+    IntegratedPayment.options.form.exp = integratedPaymentApi.expiration(
+      document.querySelector('.exp-input-container'),
+      {
+        default: IntegratedPayment.options.inputStyle.default,
+        placeholder: payplug_integrated_payment_params.exp
+      }
+    );
+    IntegratedPayment.options.schemes = integratedPaymentApi.getSupportedSchemes();
+    IntegratedPayment.bindEvents();
+    IntegratedPayment.fieldValidation();
+  },
+  bindEvents() {
+    document.querySelector('#paid').addEventListener('click', (event) => {
+      event.preventDefault();
+      integratedPaymentApi.validateForm();
+    });
+    integratedPaymentApi.onValidateForm(async ({isFormValid}) => {
+      if (isFormValid) {
+        if (payplug_integrated_payment_params.payment_id !== undefined) {
+          integratedPaymentApi.pay(payplug_integrated_payment_params.payment_id, 'visa');
+          return;
+        }
+        IntegratedPayment.options.save_card = document.querySelector('[name=schemeOptions]:checked').value;
+        IntegratedPayment.options.scheme = document.querySelector('#savecard').checked;
+
+        const response = await fetch(payplug_integrated_payment_params.routes.init_payment, {method: 'POST'});
+        const data = await response.json();
+        integratedPaymentApi.pay(data.payment_id, IntegratedPayment.options.scheme, {save_card: IntegratedPayment.options.save_card});
+      } else {
+        console.log(isFormValid);
```

Comment:
Maybe a better log ? I guess it will only log "false" here, no ?

Reply:
yup this has to be cleaned